diff --git a/node_modules/next-mdx-enhanced/loader.js b/node_modules/next-mdx-enhanced/loader.js
index 0117d34..d63b298 100644
--- a/node_modules/next-mdx-enhanced/loader.js
+++ b/node_modules/next-mdx-enhanced/loader.js
@@ -120,6 +120,7 @@ async function processLayout(options, frontMatter, content, resourcePath, scans)
   // Import the layout, export the layout-wrapped content, pass front matter into layout
   return `import layout from '${normalizeToUnixPath(layoutPath)}'
 
+export * from '${normalizeToUnixPath(layoutPath)}'      
 export default layout(${stringifyObject({
     ...frontMatter,
     ...extendedFm,
